74 refactor the UI for the profile page

Isabella-Tsui opened the pull request:
The following work was accomplished in this ticket:

- [x] User should still be able to log out
- [x] Users can see their avg sender/driver ratings and can see how long they've been on the app (years)
- [x] There are messages that indicate if the user does sender/driver reviews and if they have just joined the app
- [x] Users should be able to see the reviews that senders and drivers have given them

OmarMadhani commented:
Looks good!

Some things to consider: 
- Use usernames for reviewers to preserve anonymity
- Date of review?
- Says 55 years shipping experience for me
- What is displayed in place of average ratings if I've never shipped or delivered.

<img width="327" alt="Screenshot 2024-11-26 at 4 58 28 PM" src="https://github.com/user-attachments/assets/21a5b5a7-0fdc-4cfa-8b7d-d99b2e8a13bf">

Isabella-Tsui commented:
Here's what I did:

- Fixed the error that Omar was seeing
- Since users can be both senders and drivers, they should be able to see what people have said about them (as senders) and what they have written about others (as drivers). I changed the language so this is more clear. 

Of the things to consider, I only implemented the last point. Names are shown on AirBnb/DoorDash/Uber, so I left those as is. 55 years of shipping appeared because there was an issue in the database. I didn't have enough time to implement date of review.

OmarMadhani commented:
I have pushed some changes to your branch. Instead of reviews left by Omar, it should be reviews written about Omar (the sender or driver). I have modified the function to fetch reviews to consider the reviewtype attribute.

I also made it so that if a user joined this year, it shows Joined (year of profile.created_at) to avoid text-overflow
